### PR TITLE
biglybt: init at 3.5.0.0

### LINKED ISSUE #124253: Package request: BiglyBT, Vuze continuation
**Project description**
A torrent downloader written in Java, with support for almost all things related to torrent (i2p, tor, dht, mainline dht, etc. _see homepage_)

Vuze development stopped 4 years ago, BiglyBT is continuation of Vuze.

**Metadata**
* homepage URL: https://www.biglybt.com/
* source URL: https://github.com/BiglySoftware/BiglyBT
* license: GPL-2.0
* platforms: it's java - Windows, Linux, MacOS (legacy too), BSD, Android ... _(it should probably work even on haiku)

### PR DESCRIPTION
**Pull request by raspher**
## Description of changes

fixes: #124253 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by raspher** [comment]
Current state: It isn't perfect but it's working. Currently it's creating plugins directory in pwd. This could be patched to TMP/TEMP directory inside this project.

From their github
> Working Directory should be a new folder, since the app will write things to it. Put the [aereg.dll](https://github.com/BiglySoftware/BiglyBT/blob/master/core/lib/libWIN32Access/README.md) or libOSXAccess.jnilib in there

Should i use magic trick inside wrapper or patch this in the source even before compilation?

**Comment by superherointj** [comment]
Can you package it from source?

There is a [maven.buildMavenPackage abstraction](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/maven.section.md).

**Comment by raspher** [comment]
I've tried initially, got some problems with plugins but I'll try again from source

**Comment by superherointj** [comment]
A random idea: Consider the possibility of `plugins` being part of application [and not source application].
In case, you are still unable to build from sources. We can go ahead and use upstream binary.

File: pkgs/by-name/bi/biglybt/package.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      --prefix PATH : ${lib.makeBinPath [ jre ]}
+
+    ln -s $out/share/biglybt/biglybt $out/bin/
+    runHook postInstall
+  '';
+
+  passthru.updateScript.command = [ ./update.sh ];
+
+  meta = with lib; {
+    description = "Feature-filled Bittorrent client based on the Azureus";
```
Comment (superherointj) [inline pkgs/by-name/bi/biglybt/package.nix:49]:
```suggestion
    description = "Feature-filled Bittorrent client based on the Azureus";
    downloadPage = "https://github.com/BiglySoftware/BiglyBT";
```

Comment (superherointj) [reply pkgs/by-name/bi/biglybt/package.nix:49]: Please consider adding yourself as maintainer of this package.

File: pkgs/by-name/bi/biglybt/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    wrapProgram $out/share/biglybt/biglybt \
+      --prefix PATH : ${lib.makeBinPath [ jre ]}
+
+    ln -s $out/share/biglybt/biglybt $out/bin/
+    runHook postInstall
+  '';
+
+  passthru.updateScript.command = [ ./update.sh ];
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/bi/biglybt/package.nix:48]:
```suggestion
  meta = {
```

File: pkgs/by-name/bi/biglybt/package.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    description = "Feature-filled Bittorrent client based on the Azureus";
+    homepage = "https://www.biglybt.com/";
+    license = licenses.gpl3Plus;
```
Comment (superherointj) [inline pkgs/by-name/bi/biglybt/package.nix:51]:
```suggestion
    license = lib.licenses.gpl3Plus;
```

File: pkgs/by-name/bi/biglybt/package.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    platforms = platforms.unix;
```
Comment (superherointj) [inline pkgs/by-name/bi/biglybt/package.nix:52]:
```suggestion
    platforms = lib.platforms.unix;
```

**Comment by superherointj** [comment]
ping @raspher

**Comment by superherointj** [comment]
raspher, if you wish, we can go ahead and work in the PR as a binary. And then, whenever you find a source based solution have it replaced.

**Comment by raspher** [comment]
It would be a good idea as I'm recently a little busy at work and personal life. If someone has a working package from source please feel free to make pr.

**Comment by superherointj** [comment]
Thanks. I'm trying to figure out why `nixpkgs-review` is failing for me. It might be on my side.

**Comment by raspher** [comment]
Fixed formatting as requested. It builds and runs fine on my side, sorry for so much pushes, i'm a a bit distracted and was setting nix language server in meantime. Now it's ready for review.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 287104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>biglybt</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

**Pull request merged**